pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -477,6 +474,7 @@ func (s *ServiceEntryStore) WorkloadInstanceHandler(wi *model.WorkloadInstance,
 	} else {
 		s.serviceInstances.updateInstances(key, instances)
 	}
+	s.mutex.Unlock()
 
 	s.edsUpdate(instances, true)
```

Comment:
Not sure I understand this change. I thought the whole point of the EDS was to ensure that we process EDS events in order. But now we release the lock **then** enqueue the update -- so we lose strict ordering?

Reply:
The order is guaranteed by enqueueing, in the function, the instances enqueue by order. So they are processed in strict order, I think

Reply:
But between mutex unlock and s.edsUpdate, anything can happen. In extreme scenario the thread is not live for 10s, and 100s of events are processed in this time. Then finally we wake back up and wipe out the old configs?

Reply:
In this case, the service instances store can be updated to the latest state. And in `edsUpdate`, the eds cache is then updated to latest state. I could not think how it can not work.

Reply:
ok, I think I misunderstood. This is just telling it the keys it needs to update. The rest happens under a lock